### ISSUE TITLE
What platforms should Readium-2 support?

### ISSUE THREAD

**Issue by rkwright** [readium/readium-2#1]
#### This issue is a Discussion

What platforms should be supported?

**iOS**
-  Minimum version?

**Android**
-  Minimum version?

**OSX**
-  Minimum version?

**Windows**
-  Minimum version?
-  Desktop vs. tablet vs. phone?

**Linux**
-  Which distribution?
-  Which version?

**Chrome OS?**
-  Which version?

**Chromebooks?**

**Comment by rkwright** [comment]
[Jean-Marie Geoffroy] ere Android Support - Top 10 Android versions for "Books and references"

| Version | Mkt share | Chromium-based WebView |
|---------|-----------|-----------------------------|
|4.4 | 29.97% | Y
|5.0 | 12.71% | Y
|6.0 |12.35%|Y|
|4.1| 11.75% | N |
|5.1|10.09%|Y|
|4.2|9.90%|N|
|4.0.3 - 4.0.4|4.87%|N|
|2.3.3 - 2.3.7|4.49%|N|
|4.3|3.83%|N|
|2.2|0.54%|N|

**Comment by rkwright** [comment]
From Daniel Weck:
@JMG  re can bundle a Chromium-based WebView in their app:
e.g. https://crosswalk-project.org/

**Comment by rkwright** [comment]
From Jean-Marie.  @dweck:
Yes, this is the one we do embed. We have a few issues with it however, but not related to rendering itself: a lot of crashes, apparently caused by interferences between its native code and the standard webview native code (we need to further investigate, BTW. This is just a suggestion from the XWalk team).

**Comment by rkwright** [comment]
From James English:
Over 50% of internet traffic is on mobile
89% of mobile internet traffic is through an Application (not mobile web)
OS Global Marketshare:
Android 86.2%
iOS 12.9%
MS .6%
https://www.statista.com/statistics/266136/global-market-share-held-by-smartphone-operating-systems/

**Comment by rkwright** [comment]
From Yuri Khramov:
**re Windows version:**
Win 10 universal app should deliver all of that. I would suggest that we drop anything below Win 10.

**re Linux **'
Is there really a need for it? I believe the market is too fragmented to justify the efforts.

**re Chromebooks**
Yes and No. We face many problems with Chromebooks, and I would be happy if Readium would skip it. However, many customers keep coming with the request to support them.

**Comment by olivierkorner** [comment]
For iOS, version 7 or 8 should be a minimal target. In Xcode 8, the latest version to date, you can't select a deployment target below 8.

**Comment by tddpirate** [comment]
I suggest that for Linux support, choose one platform (say, Debian Jessie or Ubuntu 14.04) and ensure that Readium can work well inside a Docker container.
This is, more or less, the approach taken by applications like Android Studio.

**Comment by rkwright** [comment]
## Proposed Solution:
Per the discussion at the readium-2 meeting on 5 October, the consensus in the meeting was that the platform strategy would consist of supporting the following platforms:

| Platform | Min version | Comment                                |
|---------|-------------|------------------------------|
| iOS        | 8                  | cf. comment by Olivier above |
| Android | 4.4               | Kit-Kat                                      |
| OSX       | TBD             | 9?                                              |
| Win        | 10                |                                                   |
| Linux     | TBD             | Support for Linux still TBD      |

1. In this context, support means that the code, when compiled into an app and run, will pass the EPUB Test Suite on the specified platform.  Note that due to the agile nature of the Readium-2 effort, some functionality such as media-overlays may not be present in the initial builds, but all basic functionality (parsing, rendering etc.) should be present and functioning.  However, such applications may or may not run reliably on the above platforms below the minimum versions listed above. 
2. It is assumed at this point in the planning that the strategy to support desktops will be based on a JavaScript implementation wrapped in an Electron shell with hooks to native functionality to support DRM.  The same JavaScript stack  will be used to replace the CloudReader and the Chrome Extension (packaged app).
3. Linux support is still a TBD.  The proposed Electron app should run without problems on Linux (distribution TBD) however, testing more problematic.  This will probably be postponed to later in the Readium-2 cycle unless some serious volunteer steps forward.

**Comment by zhouzichao0709** [comment]
Hi guys, saw your comments and notes about the different platform strategies. I have one question regarding Android. On the summary page it is saying Kotlin version of Readium-2 will be ported from Swift. Is there a repo where that is happening? Or is it happening?

I assume https://github.com/readium/SDKLauncher-Android is Readium 1? 

Any insight or help is appreciated!

**Comment by llemeurfr** [comment]
Hi, the port is currently done in EDRLab, in parallel with the progression of the Swift development (which is a public project for a few months now). 
The project will be made public on the Readium repo in a few weeks, as soon as an alpha version can be exposed. As soon as it is public, we would be happy to get help from a new community around this new reading engine. 

The "Readium 1" android stack is made of the Readium SDK + Readium shared js. The android launcher is an example of how an app can be constructed out of these two modules.

**Comment by zhouzichao0709** [comment]
Sound good! That is very good news! Thanks for the insights! Hope to contribute more once it comes out.

**Comment by JayPanoz** [comment]
Quick question as it kinda is a pita at the ReadiumCSS level right now…

Does Windows 10 imply Internet Explorer 11 or can we skip it? 

Mind you, according to caniuse, [IE11 is more popular than MS Edge](https://caniuse.com/#feat=css-variables) but quite frankly, Microsoft is planning to [support it until October 14, 2025](http://www.allyncs.com/docs/lifecyclesupport.html) and it’s an awful lot of additional work for a 2.78% global usage which is meant to fall over this time period – and maybe R2 usage will be lower, I don’t know.

Since we skipped Android < 5, it would make sense to skip IE 11 as well. WDYT?

**Comment by llemeurfr** [comment]
We have chosen to adopt modern and evolutive Web technologies: it would be embarrassing to modify our architectural choices for the sake of preparing a not yet developed Readium-2 Web application to be compatible with a deprecated browser. The Readium Cloud Reader (aka ReadiumJS) works well with IE 11, let's use it when needed (I'm sure that the activation of Readium-2 Web vs ReadiumJS can be triggered by the browser type in most situations).
